worlds/oot_soh/location_access/overworld/goron_city.py

```diff
@@ -6,7 +6,141 @@
+        (EventLocations.GORON_CITY_STICK_POT, Events.CAN_FARM_STICKS, lambda bundle: is_child(bundle)),
+        (EventLocations.GORON_CITY_BUG_ROCK, Events.CAN_ACCESS_BUGS, lambda bundle: blast_or_smash(bundle) or can_use(Items.SILVER_GAUNTLETS, bundle)),
+        (EventLocations.GORON_CITY_CHILD_FIRE, LocalEvents.GORON_CITY_CHILD_FIRE_LIT, lambda bundle: is_child(bundle) and can_use(Items.DINS_FIRE, bundle)),
+        (EventLocations.GORON_CITY_WOODS_WARP, LocalEvents.GORON_CITY_WOODS_WARP_OPEN, lambda bundle: can_detonate_upright_bomb_flower(bundle) or can_use(Items.MEGATON_HAMMER, bundle) or has_item(LocalEvents.GORON_CITY_CHILD_FIRE_LIT, bundle)),
+        (EventLocations.GORON_CITY_DARUNIAS_DOOR_AS_CHILD, LocalEvents.GORON_CITY_DARUNIAS_DOOR_OPENED_AS_CHILD, lambda bundle: is_child(bundle) and can_use(Items.ZELDAS_LULLABY, bundle)),
+        (EventLocations.GORON_CITY_STOP_ROLLING_GORON_AS_ADULT, LocalEvents.GORON_CITY_STOP_ROLLING_GORON_AS_ADULT, lambda bundle: is_adult(bundle) and (has_item(Items.GORONS_BRACELET, bundle) or has_explosives(bundle) or can_use(Items.FAIRY_BOW, bundle) or (can_do_trick(Tricks.GC_LINK_GORON_DINS, bundle) and (can_use(Items.DINS_FIRE, bundle) or (can_do_trick(Tricks.BLUE_FIRE_MUD_WALLS, bundle) and can_use(Items.BOTTLE_WITH_BLUE_FIRE, bundle))))))
+    ])
+    # Locations
+    add_locations(Regions.GORON_CITY, world, [
+        (Locations.GC_MAZE_LEFT_CHEST, lambda bundle: can_use_any([Items.MEGATON_HAMMER, Items.SILVER_GAUNTLETS], bundle) or (can_do_trick(Tricks.GC_LEFTMOST, bundle) and has_explosives(bundle) and can_use(Items.HOVER_BOOTS, bundle))),
+        (Locations.GC_MAZE_CENTER_CHEST, lambda bundle: blast_or_smash(bundle) or can_use(Items.SILVER_GAUNTLETS, bundle)),
+        (Locations.GC_MAZE_RIGHT_CHEST, lambda bundle: blast_or_smash(bundle) or can_use(Items.SILVER_GAUNTLETS, bundle)),
+        (Locations.GC_POT_FREESTANDING_PO_H, lambda bundle: is_child(bundle) and has_item(LocalEvents.GORON_CITY_CHILD_FIRE_LIT, bundle) and (can_use(Items.BOMB_BAG, bundle) or (has_item(Items.GORONS_BRACELET, bundle) and can_do_trick(Tricks.GC_POT_STRENGTH)) or (can_use(Items.BOMBCHUS_5, bundle) and can_do_trick(Tricks.GC_POT, bundle)))),
+        (Locations.GC_ROLLING_GORON_AS_CHILD, lambda bundle: is_child(bundle) and (has_explosives(bundle) or (has_item(Items.GORONS_BRACELET, bundle) and can_do_trick(Tricks.GC_ROLLING_STRENGTH, bundle)))),
+        (Locations.GC_ROLLING_GORON_AS_ADULT, lambda bundle: has_item(LocalEvents.GORON_CITY_STOP_ROLLING_GORON_AS_ADULT, bundle)),
+        (Locations.GC_GS_BOULDER_MAZE, lambda bundle: is_child(bundle) and blast_or_smash(bundle)),
+        (Locations.GC_GS_CENTER_PLATFORM, lambda bundle: is_adult(bundle) and can_attack(bundle)),
+        (Locations.GC_MEDIGORON, lambda bundle: is_adult(bundle) and (can_break_mud_walls(bundle) or has_item(Items.GORONS_BRACELET, bundle))),
```

Comment:
You couldn't have known this, but `GC_MEDIGORON` needs a check for adult wallet too (ship handles randomized shop items differently somehow)